src/server/server.ts

```diff
@@ -173,17 +184,62 @@ module ts.server {
             });
 
             rl.on('close',() => {
-                this.projectService.closeLog();
                 this.projectService.log("Exiting...");
+                this.projectService.closeLog();
                 process.exit(0);
             });
         }
     }
 
+    interface LogEnv {
+        file?: string;
+        level?: string;
+    }
+
+    function parseLogEnv(logEnvStr: string): LogEnv {
+        var logEnv: LogEnv = {};
+        var args = logEnvStr.split(' ');
+        for (var i = 0, len = args.length; i < len; i++) {
+            var option = args[i];
+            var value = args[i + 1];
+            if (option && value) {
+                switch (option) {
+                    case "-file":
+                        logEnv.file = value;
+                        break;
+                    case "-level":
+                        logEnv.level = value;
+                        break;
+                }
+            }
+        }
+        return logEnv;
+    }
+
+    // TSS_LOG "{ level: "normal | verbose | terse", file?: string}"
+    function createLoggerFromEnv() {
+        var fileName: string = undefined;
+        var level = "normal";
+        var logEnvStr = process.env["TSS_LOG"];
+        if (logEnvStr) {
+            var logEnv = parseLogEnv(logEnvStr);
+            if (logEnv.file) {
+                fileName = logEnv.file;
+            }
+            else {
+                fileName = __dirname + "/.log" + process.pid.toString();
+            }
+            if (logEnv.level) {
+                level = logEnv.level;
+            }
+        }
+        var logger = new Logger(fileName, level);
+        return logger;

```

Comment:
I'd say just return the new logger unless you're gaining some level of debuggability.